python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -47,9 +50,23 @@ def __init__(
                 The actual rows passed may be less if the dataset is small.
             ray_remote_args: Customize the ray remote args for this op's tasks.
         """
+        ray_remote_args = _canonicalize_ray_remote_args(ray_remote_args or {})
         compute_strategy = compute_strategy or TaskPoolStrategy()
+        if isinstance(compute_strategy, TaskPoolStrategy):
+            self._base_resource_usage = ExecutionResources()
+        elif isinstance(compute_strategy, ActorPoolStrategy):
+            self._base_resource_usage = ExecutionResources(
+                cpu=ray_remote_args.get("num_cpus", 0) * compute_strategy.min_size,
+                gpu=ray_remote_args.get("num_gpus", 0) * compute_strategy.min_size,
+            )
```

Comment:
Is there a way that we could push this difference down into the task pool and actor submitters, so we don't have to do this compute strategy switching here? I was hoping to isolate all of the differences between the two compute strategies to those task submitters.